protos/sigstore_common.proto

```diff
@@ -60,13 +62,39 @@ message MessageSignature {
         bytes signature = 2;
 }
 
+// LogId captures the identity of a transparency log.
+message LogId {
+        oneof id {
+                // The unique id of the log, represented as the SHA-256 hash
+                // of the log's public key, computed over the DER encoding.
+                // https://www.rfc-editor.org/rfc/rfc6962#section-3.2
+                bytes key_id = 1;
+                // Currently not used but proposed by
+                // https://datatracker.ietf.org/doc/rfc9162/
+                string oid = 2;
+        }
+}
+
 // This message holds a RFC 3161 timestamp.
 message RFC3161SignedTimestamp {
         // Signed timestamp is the DER encoded TimeStampResponse.
         // See https://www.rfc-editor.org/rfc/rfc3161.html#section-2.4.2
         bytes signed_timestamp = 1;
 }
 
+message PublicKey {
+        // Optional key identifier. Format must be agreed upon out of band
+        // by signer and verifier. See PublicKeyIdentifier for more
+        // information.
+        // Implementors MAY provide a mapping function from hint to PublicKey.
+        string key_id = 1;
+        // DER-encoded public key.
+        bytes raw_bytes = 2;
+        // The time range is half-open and does not include the end
+        // timestamp, i.e [valid_for.start, valid_for.end).
```

Comment:
What if it doesn't have an end time, as in if it's the current key?

Reply:
same here: https://github.com/sigstore/protobuf-specs/pull/5/files#r1010726774